DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Models;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class RegisterController : Controller
+    {
+        private const string CookieName = "RegistrationData";
+        private readonly ICentresService centresService;
+        private readonly IJobGroupsService jobGroupsService;
+
+        public RegisterController(ICentresService centresService, IJobGroupsService jobGroupsService)
+        {
+            this.centresService = centresService;
+            this.jobGroupsService = jobGroupsService;
+        }
+
+        public IActionResult Index()
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            var delegateRegistrationData = TempData.Get<DelegateRegistrationData>();
+
+            if (delegateRegistrationData == null || !Request.Cookies.ContainsKey(CookieName))

```

Comment:
VS seems to think that this cannot be null - could we fix the return type in TempDataExtension?

Reply:
Rider has the same warning and I forgot to fix it 🙄
Making the return type nullable clears it all up